### PR TITLE
Define separate SignedStatement & UnsignedStatement classes

### PR DESCRIPTION
**Pull request by yusefnapora**
This refactors the Statement class hierarchy a bit to make a clean split between signed and unsigned statements.

There's now a base `Statement` class that has everything except the `signature` field, plus common helper methods like `.toProtobuf`, `.expandObjects`, etc.

The `UnsignedStatement` subclass adds the `.calculateSignature` and `.sign` methods, which return `Buffer` and `SignedStatement`, respectively.

The `SignedStatement` class has the `signature` field and the `.verifySignature` method.  Both `SignedStatement` and `UnsignedStatement` inherit from `Statement` directly, so `SignedStatement` does not have the `.sign` and `.calculateSignature` methods.  However, there's a helper on the the base class `.asUnsignedStatement()` that will return an unsigned copy of either a signed or unsigned statement.  So, if you need to strip an existing signature and re-sign a message, you can do:

```javascript
mySignedStmt.asUnsignedStatement().sign(newPublisherId)
  .then(signed => {
    // `signed` has a new signature
  })
```

Converting from protobuf still happens with `Statement.fromProtobuf()`, which will return a `SignedStatement` if the protobuf message has a `signature`, and an `UnsignedStatement` otherwise.

### CODE REVIEW

File: src/model/statement.js [hunk 1]
```diff
@@ -14,72 +14,29 @@ class Statement {
...
-    return Statement.fromProtobuf(pb.stmt.Statement.decode(msgBuffer))
+  static fromBytes (bytes: Buffer): Statement {
+    return Statement.fromProtobuf(pb.stmt.Statement.decode(bytes))
   }
 
   toProtobuf (): StatementMsg {
-    const {id, publisher, namespace, timestamp, signature, body} = this
-    return {id, publisher, namespace, timestamp: timestamp, signature, body: body.toProtobuf()}
+    const {id, publisher, namespace, timestamp, body} = this
+    const msg: StatementMsg = {id, publisher, namespace, timestamp: timestamp, body: body.toProtobuf()}
```
Comment (parkan) [inline src/model/statement.js]: spent a while staring at these before concurring that this is in fact the best way to do it w/o libs

Comment (yusefnapora) [inline src/model/statement.js]: haha, yeah.  It does look pretty odd :)

Comment (yusefnapora) [inline src/model/statement.js]: actually, I don't know why I did `timestamp: timestamp` instead of just `timestamp` - probably some earlier refactor.

Comment (parkan) [inline src/model/statement.js]: oh yeah, that one can get short-handed

**Pull request merged**